themes/ddbasic/README.md

```diff
@@ -1,18 +1,206 @@
-# DDBasic theme for Ding2 V2.0
-#
-#
+# DDBasic V2.0
 
-## Maintenance pages
+DDBasic is the default theme provided with DDBCMS and the recommended parent theme for inheritance of DDBCMS subthemes.
 
-The theme provides maintenance page templates:
-- *maintenance-page-tpl.php*: This page template is used when the site has been set offline by the administrator. 
-- *maintenance-page--offline.tpl.php*: This page template is used when the site cannot get in contact with the database.
+This document explains important parts of the theme layer and best practices for theming modules and inheriting subthemes from DDBasic.
 
-It is common for ding2 sites to have the DDBasic theme reside in the folder /profiles/ding2/themes/ddbasic.
-In this case, you will need to add the following lines to your settings.php file in order for both template 
-files to be displayed properly:
+*As the update has not yet been merged into the [ding2 master branch](https://github.com/ding2/ding2) links below will refer to a temporary git repo and KSS stylesheets.*
 
-    $conf['install_profile'] = 'ding2';
-    $conf['maintenance_theme'] = 'ddbasic';
+## The DDBasic framework
 
-    
\ No newline at end of file
+DDBasic is the actual implementation of the frontend for DDBCMS, and can be considered a kind of framework that works in many layers:
+
+* Drupal: view modes
+* Drupal: image styles
+* html : markup and classes with special purposes
+* scss : mixins, variables, extensions and responsive styling
+* js : helper and ui functions
+* php : Drupal template functions and other similar functions
+* gulp: Developer tools
+
+To reduce duplication of functionality and styling and to maintain consistence of both implementation and user experience, DDBasic provides a documentation for each of these layers.
+
+## View modes
+
+Drupal view modes allows for reuse of markup, styling and interaction of entities across views, panels and many other ways of rendering Drupal entities.
+
+DDBasic introduces the following basic view modes which should be used across all submodules:
+
+node:
+
+* full
+* teaser
+* teaser_highlight
+* search_result
+
+ting_object:
+
+* full
+* teaser
+* search_result
+
+Note that each view mode can be styled individually for specific purposes, and fields can perfectly valid be hidden with display:none to create simpler outputs of each view mode. 
+
+Hiding a few fields with CSS is preferred to increasing the number of view modes. 
+
+Rendered view modes is the preferred method of presentation of entities as opposed to fielded views. 
+
+**Rendering ting materials**
+
+A recurring frontend task is to present a list of materials. The prefered procedure is to render an array of material ids as a carousel:
+
+```php
+theme(‘ting_carousel’, array( ‘items’ => array(id1,id2,id3) ) );
```

Comment:
This is not valid code. Note the `‘’` instead of `''`. There should also be a space after each comma.